modules/core/src/lib/layer.ts

```diff
@@ -230,20 +230,20 @@ export default abstract class Layer<PropsT extends {} = {}> extends Component<
       coordinateSystem: this.props.coordinateSystem
     });
     const [x, y, z] = worldToPixels(worldPosition, viewport.pixelProjectionMatrix);
-    return xyz.length === 2 ? [x, y] : [x, y, z];
+    return [x, y, z];
```

Comment:
I consider this a breaking API change. Could `xyz` be typed as `[number, number] | [number, number, number]`?

Not saying yet that we can't make this API change, but generally we'll tradeoff type safety to preserve the API, and not the other way around.

Reply:
Even that will be breaking change as many people out there are likely passing a `number[]` and will need to add a cast. I would suggest using a function overload, something like:

```ts
function project(xyz: [number, number]): [number, number];
function project(xyz: [number, number, number]): [number, number, number];
function project(xyz: Vector2Or3): Vector2Or3 {
```

Reply:
Also we haven't got around to using them everywhere yet, but we have [helper types](https://github.com/visgl/deck.gl/blob/master/modules/aggregation-layers/src/common/aggregator/gpu-aggregator/bin-sorter-uniforms.ts#L5) like `NumberArray2` that you could use here